docs/What-is-Endianness.rst

```diff
@@ -0,0 +1,52 @@
+What is Endianness
+##################
+
+Endianness [#Endianness]_, at least as far as computing is concerned, is the ordering of bytes within a binary representation of data. The most common representations of endianness are :ref:`big-endian` and :ref:`little-endian`. There exist other forms, such as Middle-Endian, but those are beyond the scope of this document.
+
+.. figure:: /img/endianpig.png
+    :align: center
+    :alt: alternate text
+    :target: http://geek-and-poke.com/geekandpoke/2011/9/7/simply-explained.html
+
+    `"Simply Explained" <http://geek-and-poke.com/geekandpoke/2011/9/7/simply-explained.html>`_
+
+    Comic by `Oliver Widder <http://geek-and-poke.com/>`_ Released under `Attribution 3.0 Unported (CC BY 3.0) <https://creativecommons.org/licenses/by/3.0/>`_.
+
+Entomology
+**********
+
+The Computer Science terms Big-Endian and Little-Endian were introduced by Danny Cohen [#CohenDanny]_ in 1980. The key term *endian* has its roots in the novel Gulliver’s Travels [#GulliversTravels]_ by Jonathan Swift [#SwiftJonathan]_ where within a war occurs between two factions who are fighting over which end of a boiled egg should be opened for eating. The big end or the little end. Unsurprisingly, the same said book was the inspiration for the naming of the `Gulliver <https://github.com/sandialabs/gulliver>`_ library.
+
+.. _big-endian:
+
+Big-Endian
+**********
+
+Big-Endian, often referred to as *Network Byte Order*, ordering is left-to-right. Given the representation of an unsigned number in bytes the further to the left that a byte exists the more significant it is.
+
+For example, the decimal value of the unsigned integer ``6060842`` may be represented as ``0x5C7B2A`` in hexadecimal. This hexadecimal value is composed of the three bytes ``0x5C``, ``0x7B``, and ``0x28``. As such the value ``6060842`` may be represented in Big-Endian as a byte stream of ``[0x5C, 0x7B, 0x2A]``.
+
+Big-Endian integer representation likely comes as second nature to developers familiar with right-to-left Arabic numerals [#ArabicNumerals]_ representation.
+
+.. _little-endian:
+
+Little-Endian
+*************
+
+Little-Endian ordering is right-to-left. Given the representation of an unsigned number in bytes the further to the right a byte exists the more significant it is.
+
+For example, the decimal value of the unsigned integer ``8675309`` may be represented as ``0x85BF7D`` in hexadecimal. This hexadecimal value is composed of the three bytes ``0x85``, ``0xBF``, and ``0x7D``. As such the value ``8675309`` may be represented in Little-Endian as a byte stream of ``[0x7D, 0xBFm 0x85]``.
```

Comment:
Would it be helpful to use the same unsigned integer from the big-Endian example? That way people can “see” the difference?

Reply:
Sure, I can do that. I was just trying to switch up values to make things more "interesting". 
I'll drop the B-52's and reuse Jenny.

Reply:
Fixed in push [e3978c3cb3acfbf4bc06f74bd9518b92edabf965](https://github.com/sandialabs/Gulliver/pull/25/commits/e3978c3cb3acfbf4bc06f74bd9518b92edabf965).

Apparently I did a bad job with my Decimal to Hex conversion. That too is correctedDecimal to Hex conversion of 8675309. Also, just for fun, added latex notation to doc.